### ISSUE TITLE
Unexpected crash on MAC, unable to boot normally

### ISSUE THREAD

**Issue by Zijinn** [xor-gate/syncthing-macosx#230]
### What happened?

It can run normally only during the initial installation. After exiting and launching the installation program, it will enter a crash state when restarting: the taskbar icon flashes with an exclamation mark on it, and then crashes. The software has been set up to have full access to the disk.
Only by clicking the downloaded installation file again can you continue to enter normally.
You can also enter using the terminal version of macos, but the app cannot.

### Syncthing version

v1.28.0

### Platform & operating system

Macos Sequoia15.1 m4

**Comment by xor-gate** [comment]
@Zijinn you have probably started first time from the DMG file mounted in disk. You could probably resolve the problem according to almost same issue here: https://github.com/syncthing/syncthing-macos/issues/228. I need to find a way people using it incorrectly or not used to macOS way of working...

Please let me know if it works or not with information from the other issue.

**Comment by xor-gate** [comment]
@calmh thanks for migrating this issue, it is probably a well know usability problem. Not 100% a bug, but we should be resilient against improper use as far as possible.

**Comment by Zijinn** [comment]
@xor-gate Thanks for your reply, I re-tested following your instructions in #228 and it worked fine. If possible, I hope this can be simplified. The specific operations are as follows:

1. To check the quarantine flag, run the following command in the terminal:
`xattr -lr /Applications/<YourApp>.app`

2. If you see com.apple.quarantine in the output, use the following command to remove the quarantine flag:
`sudo xattr -dr com.apple.quarantine /Applications/<YourApp>.app`

3. Type the command in Terminal.app, then exit and start Syncthing from /Applications again:
`defaults delete com.github.xor-gate.syncthing-macosx Executable`

**Comment by xor-gate** [comment]
Sorry for my late reply, thanks for this extra commands. But I have no clue why macOS would put it in Quarantine... Did your macOS put it into quarantine?

**Comment by Zijinn** [comment]
@xor-gate After executing these commands, it can run normally. It is reasonable to speculate that macOS has isolated it, but I can't seem to find the reason for the isolation. However, this problem did not occur on the previous Intel macOS. This problem occurred on the latest Mini M4, which is also the latest version of the system.

**Comment by xor-gate** [comment]
Yeah I think the trigger was it crashed, then it was put into quarantine. Did you tried multiple times ? Or just after one crash it was put into quarantine?

**Comment by Zijinn** [comment]
Yes, I tried reinstalling and booting multiple times. Only during the first installation, the installation package has not yet popped up and can run normally. Once it pops up, it won't run, even though I set it up to allow access to the disk.

**Comment by kratoz29** [comment]
I am getting a crash on the [latest](https://github.com/syncthing/syncthing-macos/releases/tag/v2.0.14-1) version as well with Big Sur, previous versions were working fine.

**Comment by xor-gate** [comment]
> I am getting a crash on the [latest](https://github.com/syncthing/syncthing-macos/releases/tag/v2.0.14-1) version as well with Big Sur, previous versions were working fine.

You need at least macOS 12 with v2

**Comment by kratoz29** [comment]
> > I am getting a crash on the [latest](https://github.com/syncthing/syncthing-macos/releases/tag/v2.0.14-1) version as well with Big Sur, previous versions were working fine.
> 
> You need at least macOS 12 with v2

You are completely right, I overlooked that, I rolled back for now...
